pkg/vm/memory/memory.go

```diff
@@ -8,24 +8,67 @@ import (
 
 // Represents a write-once Memory Cell
 type Cell struct {
-	Value    f.Element
+	Value    *MemoryValue
 	Accessed bool
 }
 
-func (cell *Cell) Read() f.Element {
-	cell.Accessed = true
-	return cell.Value
+type Segment struct {
+	Data map[f.Element]Cell
```

Comment:
Is there a benefit of using a separate type for segments? Rather than modelling memory as ```map[MemoryAddress]*MemoryValue```?

Reply:
That's a fair point, though I don't know why it shows you the old code. We are currently using an array insted of a map

Reply:
The reason behind Segment was to make it more readable, like the Segment word make it really obvious what you are dealing with. I think it is a zero cost abstraction as well. But we can change it